### PR TITLE
Configure texture usage for Swap Chains

### PR DESCRIPTION
**Pull request by kainino0x**
Resurrection of #118

### CODE REVIEW

**Comment by kainino0x** [comment]
I'll add the last few comments from the original review soon and send this for review.

Not sure how to fix the CLA issue. I guess the easiest is to either `--reset-author` or admin-force-merge.

**Comment by Kangz** [comment]
I'd rather not admin-force merge, Austin's contribution was done while at Google so we can reset-author without issues.

**Comment by kainino0x** [comment]
Answering remaining questions on the original PR.

> @kainino0x: didn't we say that when the next texture is provided it should be in the None usage so we force a transition? This would remove the need for initialUsage.

I removed initialUsage in my last commit but I'm not set on that decision. @Kangz WDYT?

> The D3D12Binding still does a transition when it swaps buffers. Should this be removed/moved to BeginRenderSubpass?

I removed it.

> Actually, now that I think of it, should we only allow pure write formats for swapchain textures so the application cannot see garbage? Or should we allow any format but have an implicit clear?

We decided to have an implicit clear (but I have not inserted it).

**Comment by kainino0x** [comment]
Hey @Kangz, I realized this is still floating around. PTAL.

File: src/utils/D3D12Binding.cpp [hunk 1]
```diff
@@ -77,6 +77,41 @@ namespace utils {
 
             return factory;
         }
+
+        DXGI_USAGE D3D12SwapChainBufferUsage(nxtTextureUsageBit allowedUsages) {
```
Comment (Kangz) [inline src/utils/D3D12Binding.cpp]: Eventually we'll need to find a way to factor this. No need to address now.

**Pull request merged**